system/ee/installer/updates/ud_7_04_00.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * This source file is part of the open source project
+ * ExpressionEngine (https://expressionengine.com)
+ *
+ * @link      https://expressionengine.com/
+ * @copyright Copyright (c) 2003-2023, Packet Tide, LLC (https://www.packettide.com)
+ * @license   https://expressionengine.com/license Licensed under Apache License, Version 2.0
+ */
+
+namespace ExpressionEngine\Updater\Version_7_4_0;
+
+/**
+ * Update
+ */
+class Updater
+{
+    public $version_suffix = '';
+
+    /**
+     * Do Update
+     *
+     * @return TRUE
+     */
+    public function do_update()
+    {
+        $steps = new \ProgressIterator(
+            [
+                'ensureBuiltinRoles',
+            ]
+        );
+
+        foreach ($steps as $k => $v) {
+            $this->$v();
+        }
+
+        return true;
+    }
+
+    // in some very old EE versions is was possible to delete built-in member groups
+    // here we make sure the required roles are in place
+    private function ensureBuiltinRoles()
+    {
+        $rolesQuery = ee('Model')->get('Role')->fields('role_id')->filter('role_id', 'IN', [1, 2, 3, 4, 5]);
```

Comment:
@intoeetive Is it necessary for the built-in roles to have these `role_id`s 1-5?  Or should we be looking to ensure the role `short_name`s are populated and use whatever role_ids are available in the case that custom roles have been created?  Or were there always role_ids for 1-5 and any custom roles would have started at id 6 so this is safe to do?

Reply:
Yes, the numbers are necessary (in fact, we even have them as constants)